SQL_Analysis_Nino_Jintcharadze_WindFunc_HW.sql

```diff
@@ -0,0 +1,124 @@
+-- Task 1: Calculate total sales per channel and customer sales with sales percentage
+
+WITH TotalSales AS (
+    -- Calculate total sales per channel
+    SELECT 
+        channel_id,
+        SUM(amount_sold) AS total_sales
+    FROM 
+        sh.sales
+    GROUP BY 
+        channel_id  -- Group by channel to get total sales for each channel
+),
+CustomerSales AS (
+    -- Calculate total sales for each customer per channel
+    SELECT 
+        s.channel_id,
+        c.cust_last_name,
+        c.cust_first_name,
+        SUM(s.amount_sold) AS amount_sold
+    FROM 
+        sh.sales s
+    JOIN 
+        sh.customers c ON s.cust_id = c.cust_id 
+    GROUP BY 
+        s.channel_id, c.cust_last_name, c.cust_first_name  
+)
+SELECT 
+    cs.channel_id,
+    cs.cust_last_name,
+    cs.cust_first_name,
+    ROUND(cs.amount_sold, 2) AS amount_sold,  -- Round the sales amount to 2 decimal places
+    CONCAT(ROUND((cs.amount_sold / ts.total_sales) * 100, 5), ' %') AS sales_percentage  -- Calculate sales percentage
+FROM 
+    CustomerSales cs
+JOIN 
+    TotalSales ts ON cs.channel_id = ts.channel_id  -- Join to get total sales by channel
+ORDER BY 
+    cs.channel_id, cs.amount_sold DESC  -- Order by channel and sales amount
+LIMIT 5;  -- Limit the results to the top 5 customers per channel
+
+-- Task 2: Generate quarterly sales report for products in 2000 using crosstab
+
+CREATE EXTENSION IF NOT EXISTS tablefunc;  -- Enable tablefunc extension for crosstab functionality
+
+WITH SalesData AS (
+    -- Gather total sales for each product by quarter
+    SELECT 
+        p.prod_name,
+        SUM(s.amount_sold) AS total_sales,
+        EXTRACT(QUARTER FROM t.time_id) AS quarter  -- Extract quarter from time_id
+    FROM 
+        sh.sales s
+    JOIN 
+        sh.products p ON s.prod_id = p.prod_id  
+    JOIN 
+        sh.times t ON s.time_id = t.time_id  
+    WHERE 
+        t.calendar_year = 2000  
```

Comment:
add additional conditions - for all products in the Photo category in the Asian region